amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -146,7 +156,7 @@ export class SearchPage extends React.Component<SearchPageProps> {
     return {
       term: (searchTerm || '').trim(),
       tab: this.getSelectedTabByResourceType(selectedTab),
-      index: isNaN(index) ? 0 : index,
+      index: isNaN(index) ? null : index,
```

Comment:
What is the logic surrounding this needing to be `null`? 

For this case and the case above, I'm cautious of switching things to `null` for a feature or bug fix. We don't have `strictNullChecks` configured on our TypeSxript code yet, so we could be opening ourselves up for bugs where this value gets passed to into code that expects this to be a `number` -- even if that's not the case in this PR, someone could write code in the future that does. But I'd like to understand the reason first in order to think about how we might resolve that concern.

Reply:
This function reads the URL parameters. Setting the default value for `pageIndex` to `0` or `selectedTab` to the previous tab obscures the fact that they have actually been reset. The null state needs to be passed along to trigger the auto-tab and index selection.